### ISSUE TITLE
Awesome library and example... few suggestions too

### ISSUE THREAD

**Issue by sukeshak** [peakhunt/esp32_imu#1]
Awesome sample with some never seen before additions with Veu and the the poll.sh script :)

A suggestion would be to make it easy to integrate with any other LCD to the project.
Probably use EP32_TFT_library?

I am using M5stack with [ESP32_TFT_library](https://github.com/loboris/ESP32_TFT_library) which uses ILI9341.

Also can we remove mongoose dependency?
Thanks once again for the very neat implementation 👍

**Comment by sukeshak** [comment]
Just completed switching to ESP32_TFT_library for display.
There is a conflict with list related functions from spiff library and your list.h in component folder

/components/utils/include
- list.c
- list.h

(I changed here for now)
/components/spiffs 
- list.c
- list.h

**Comment by peakhunt** [comment]
right! actually the name has been causing some naming conflicts for me too.
But the list.h, which is from linux kernel source, is the best list library
I've ever seen. Maybe it's time to quit on my being lazy and change the
name to something like "generic_list.h". Done it.

thanks for your kind words for my hobby project in progress.

-H.Kim-

2018-06-01 0:13 GMT+09:00 Sukesh Ashok Kumar <notifications@github.com>:

**Comment by peakhunt** [comment]
not sure what you mean by mongoose dependency.
I used mongoose dependency to implement JSON based WEB service
implementation.
If you don't need that, all you gotta do is just get rid of webserver.c
from main/ (I believe so, if you know what I mean).

As for ILI9341, I'm not sure I have that LCD display.

-H.Kim-

2018-05-31 20:06 GMT+09:00 Sukesh Ashok Kumar <notifications@github.com>:

**Comment by sukeshak** [comment]
Well mongoose is not completely open source, so I try to avoid. Not very important in this context :)

I merged your project with EP32_TFT_library (link above) and replaced your LCD with this library since it supports the both ILI9341 and your ST7735 + more... 

I have 3 devices running the exact same firmware... but showing different values... 
(IMU portion is your code but using mahony in this example)
Any ideas or suggestions?

![imu](https://user-images.githubusercontent.com/9167668/40880060-021b7554-66c8-11e8-985e-a21d4eb583cd.jpg)

**Comment by sukeshak** [comment]
Thank you for the response :)
#1. I will get more information on this one because their data sheet does not have much info.

#2. I tried keeping the device flat when switched on and 2nd & 3rd device comes very close in values. 1st device has still issues. Will do some more testing... maybe device specific issue.

#3. 1st device is still having issues. Will test this.

Have you looked using DMP in MPU9250? 
Guess its easier and cleaner but guess fusion is only with accelerometer + gyro.

**Comment by peakhunt** [comment]
Intentionally I avoid using DMP since my goal here is to come up with fully open AHRS FILTER.

**Comment by sukeshak** [comment]
Ah that's nice... :)

Meanwhile, have you tried KST and Processing for visualization?
https://kst-plot.kde.org/video_tutorials/

Its pretty powerful and supports Serial and JSon too...
https://processing.org/
